### PR TITLE
Fixes to allow bank charge success

### PR DESCRIPTION
**Pull request by Ajibola**
There were some json inconsistencies with the paystack API and some parameters missing or miss named.

Also added a test for the charge function.

### CODE REVIEW

File: charge_test.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	if error != nil {
+		t.Error(error)
+	}
+
+	if resp["reference"] == "" {
+		t.Error("Missing transaction reference")
+	}
+}
+
+func TestChargeService_CheckPending(t *testing.T) {
```
Comment (rpip) [inline charge_test.go]: This should be `TestChargeServiceCheckPending`

File: charge_test.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package paystack
+
+import (
+	"testing"
+)
+
+func TestChargeService_Create(t *testing.T) {
```
Comment (rpip) [inline charge_test.go]: This should be `TestChargeServiceCreate`

File: charge_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		t.Error(error)
+	}
+
+	if resp["reference"] == "" {
+		t.Error("Missing charge reference")
+	}
+
+	resp2, error := c.Charge.CheckPending(resp["reference"].(string))
+	if error != nil {
+		t.Error(error)
```
Comment (rpip) [inline charge_test.go]: Please follow the error reporting conventions. 

Should be `t.Errorf("Check pending Charge returned error: %v", err)`

File: charge_test.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	charge := ChargeRequest{
+		Email:    "your_own_email_here@gmail.com",
+		Amount:   10000,
+		Bank:     &bankAccount,
+		Birthday: "1999-12-31",
+	}
+
+	resp, error := c.Charge.Create(&charge)
```
Comment (rpip) [inline charge_test.go]: Please follow the error reporting conventions.

Should be `t.Errorf("Create Charge returned error: %v", err)`

File: charge_test.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	if error != nil {
+		t.Error(error)
```
Comment (rpip) [inline charge_test.go]: Please follow the error reporting conventions.

Should be `t.Errorf("Create Charge returned error: %v", err)`

**Review by rpip — CHANGES_REQUESTED** [review]
Thanks @Ajibola. Looks good.

Just some style nitpicks - see comments.

**Comment by Ajibola** [comment]
Hi @rpip , thanks for the well written lib. Please find the changes made as suggested.

**Pull request merged**